app/assets/javascripts/provider/admin/apiconfig/services/integration.js.coffee

```diff
@@ -181,16 +182,15 @@ $(document).on 'initialize', '#proxy', ->
     link.href = url
     link.hostname
 
-  compareHttpHosts = (a, b) ->
-    extractHost(a) == extractHost(b)
+  compareHttpHosts = (input) ->
+    extractHost(input.val()) == extractHost(input.data('default'))
 
-  extractHost = (url) ->
-    link = document.createElement('a')
-    link.href = url
-    link.hostname
-
-  compareHttpHosts = (a, b) ->
-    extractHost(a) == extractHost(b)
+  inputHasSameProtocols = (input) ->
+    actualValue = document.createElement('a')
+    actualValue.href = input.val()
+    defaultValue = document.createElement('a')
+    defaultValue.href = input.data('default')
+    actualValue.protocol == defaultValue.protocol
```

Comment:
Might be better to have a list of secure protocol instead?

Also the `inputHasSameProtocols` name does not look right to me. Change it to `inputProtocolsSecure` ?

Reply:
@hallelujah 
At this moment we are not checking if the protocol is secure or not, only if it's the same as the default data attribute. I'd say, let's just rename the method in the scope of this PR.
For other changes or improvements I'd create a new task. This file has a lot of coffeescript/jQuery spaghetti code, we should refactor into a react component or at least to ES6.